mpd/twisted.py

```diff
@@ -0,0 +1,212 @@
+        escaped_name = name.replace(" ", "_")
+        setattr(cls, escaped_name, func)
+
+    def lineReceived(self, line):
+        line = line.decode('utf-8')
+        command_list = self._state and isinstance(self._state[0], list)
+        state_list = self._state[0] if command_list else self._state
+        if line.startswith(HELLO_PREFIX):
+            self.mpd_version = line[len(HELLO_PREFIX):].strip()
+            # default state idle, enter idle
+            if self._default_idle:
+                self.idle().addCallback(self._dispatch_idle_result)
+        elif line.startswith(ERROR_PREFIX):
+            error = line[len(ERROR_PREFIX):].strip()
+            if command_list:
+                state_list[0].errback(CommandError(error))
+                for state in state_list[1:-1]:
+                    state.errback(
+                        CommandListError("An earlier command failed."))
+                state_list[-1].errback(CommandListError(error))
+                del self._state[0]
+                del self._command_list_results[0]
+            else:
+                state_list.pop(0).errback(CommandError(error))
+            # XXX: reset received lines here? 
+            self._continue_idle()
+        elif line == SUCCESS or (command_list and line == NEXT):
+            state_list.pop(0).callback(self._rcvd_lines[:])
+            self._rcvd_lines = []
+            if command_list and line == SUCCESS:
+                del self._state[0]
+            self._continue_idle()
+        else:
+            self._rcvd_lines.append(line)
+
+    def _execute(self, command, args, parser):
+        # close or kill command in command list not allowed
+        if self._command_list and not callable(parser):
+            msg = "{} not allowed in command list".format(command)
+            raise CommandListError(msg)
+        # default state idle and currently in idle state, trigger noidle
+        if self._default_idle and self._idle and command != "idle":
+            self.noidle().addCallback(self._dispatch_noidle_result)
+        # write command to MPD
+        self._write_command(command, args)
+        # create command related deferred
+        deferred = defer.Deferred()
+        # extend pending result queue
+        state = self._state[-1] if self._command_list else self._state
+        state.append(deferred)
+        # NOOP is for close and kill commands
+        if parser is not self.NOOP:
+            # attach command related result parser
+            deferred.addCallback(parser)
+            # command list, attach handler for collecting command list results
+            if self._command_list:
+                deferred.addCallback(self._parse_command_list_item)
+        return deferred
+
+    def _write_command(self, command, args=[]):
+        parts = [command]
+        parts += ['"{}"'.format(escape(arg.encode('utf-8')) \
+            if isinstance(arg, unicode) else str(arg)) for arg in args]
```

Comment:
Yes, I need to fire up a python 3 instance and fix the unicode issue. Thanks for the hint.